R/data_summary.R

```diff
@@ -103,10 +103,10 @@ data_summary.data.frame <- function(x, ..., by = NULL, include_na = TRUE) {
     }
     # split data, add NA levels, if requested
     l <- lapply(x[by], function(i) {
-      if (include_na && anyNA(i)) {
-        addNA(i)
-      } else {
+      if (remove_na || !anyNA(i)) {
```

Comment:
The order of these two conditions can be switched to take advantage of short-circuiting behaviour of `||` to avoid unnecessary computation.

Reply:
You mean `if (!anyNA(i) ||remove_na) {`?
What would be the advantage?

Reply:
`remove_na` is doing something that is relevant only if `!anyNA` is `TRUE`, so in this order it's more readable because it has more logical flow considering the short-circuiting behaviour of `||`.

Reply:
> remove_na is doing something that is relevant only if !anyNA

No, it's an OR-logical, not AND.

Reply:
Ah, you are right. In my mind, `remove_na` was an operation and that's why I was confused.